bgpd/bgp_route.c

```diff
@@ -6300,6 +6308,185 @@ static void bgp_aggregate_install(
 	bgp_dest_unlock_node(dest);
 }
 
+/**
+ * Check if the current path has different MED than other known paths.
+ *
+ * \returns `true` if the MED matched the others else `false`.
+ */
+static bool bgp_aggregate_med_match(struct bgp_aggregate *aggregate,
+				    struct bgp *bgp, struct bgp_path_info *pi)
+{
+	uint32_t cur_med = bgp_med_value(pi->attr, bgp);
+
+	/* This is the first route being analyzed. */
+	if (!aggregate->med_initialized) {
+		aggregate->med_initialized = true;
+		aggregate->med_mismatched = false;
```

Comment:
Minor nit: why do we need two seperate structure member for this? can we not do with med_flag and use bitmap?

Reply:
> why do we need two seperate structure member for this? can we not do with med_flag and use bitmap?

Clarity: I think this data structure is not used often enough to justify optimizing it yet (it would also make the code more complicated). There is only one `aggregate` data structure per `aggregate-address` and I don't think it is used often.

Reply:
My commet was not about optimization but best pratice. Since this is minor we can pass on this for now.